streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/StickyTaskAssignor.java

```diff
@@ -257,12 +260,17 @@ private int sumCapacity(final Collection<ClientState> values) {
             this.pairs = new HashSet<>(maxPairs);
         }
 
-        boolean hasNewPair(final TaskId task1, final Set<TaskId> taskIds) {
+        boolean hasNewPair(final TaskId task1,
```

Comment:
This is a meta comment about having new pair: note that `hasNewPair` would only be useful when assigning standby tasks (note that we always assign active tasks to everyone first, and only start assigning standby tasks after that), so we could refactor the code to only call this function for `assignStandby(..)`

And then then in `allocateTaskWithClientCandidates` we should consider `task1.topicGroupId != taskId.topicGroupId` along with `leastLoad`.

Reply:
Another comment in for the second phase:

```
// try and assign any remaining unassigned tasks to clients that previously
// have seen the task.
```

Note that one task could be in previous standby task of multiple clients (that's why we have `Set<ID>` in the value), we should not always pick the first `client.hasUnfulfilledQuota(tasksPerThread)` and stuff to it, while we should also consider `task1.topicGroupId != taskId.topicGroupId` with its existing active tasks.

Reply:
I am not sure if those comments are addressed already. \cc @bbejeck ?